File: LinkedList/Leetcode206.java
class Leetcode206{
    /**
     * Alright, we gonna have two solutions, for both iterative and recursive.
     */
    public ListNode reverse(ListNode head){
        /**
         *  Recursive one
         */
        if(head == null || head.next == null)return head;
        ListNode last = reverse(head.next);
        head.next.next = head;
        head.next = null;
        return last;
    }

    public ListNode reverseList(ListNode head){
        if(head == null)return null;
        ListNode newHead = null;
        while(head != null){
            ListNode temp = head.next;
            head.next = newHead;
            newHead = head;
            head = temp;
        }
        return newHead;
    }
}

File: LinkedList/Python/Leetcode206.py
class Solution:
    def Leetcode206(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # Iterate
        # newHead = None
        # while head:
        #     nextNode = head.next
        #     head.next = newHead
        #     newHead = head
        #     head = nextNode
        # return newHead
        
        # Recursion
        if not head or not head.next:
            return head
        last = self.reverseList(head.next)
        head.next.next = head
        head.next = None
        return last
